packages/database/src/models/Invoice/invoicePriceListMatching.ts

```diff
@@ -0,0 +1,53 @@
+import { differenceInYears, parseISO } from 'date-fns';
+
+export const equalsIfPresent = (
+  ruleVal: string | undefined,
+  inputVal: string | undefined,
+): boolean => {
+  if (ruleVal == null) {
+    return true; // If ruleVal not present, return true
+  }
+  return ruleVal === inputVal;
+};
+
+const OPERATORS: Record<string, (a: number, b: number) => boolean> = {
+  '<': (a: number, b: number) => a < b,
+  '<=': (a: number, b: number) => a <= b,
+  '>': (a: number, b: number) => a > b,
+  '>=': (a: number, b: number) => a >= b,
+  '=': (a: number, b: number) => a === b,
+};
+
+/**
+ * @param condition - Age condition string (e.g., ">=18", "<65", "=21")
+ * @param dob - patient date of birth as ISO date string (e.g., "2010-10-15")
+ */
+export const matchesAgeIfPresent = (
```

Comment:
I think we might have a bit of a problem with the way this is set up currently. 

In FSM we need 3 separate lists for 0-14, 15-64, 65+
If we 3 lists: 
  1: `age: <=14`
  2: `age: undefined`
  3: `age: >= 65`

Then anyone who's 14 or younger, or 65+ will match with list 2 as well as their respective list, which throws an error. We need list 2 to be able to have a filter saying `> 14 AND < 65`

I'm a big fan of expressions, but this is a case I might be leaning towards just using an object to represent this? eg.
  1: `age: { max: 14 }`
  2: `age: { min: 15, max: 64 }`
  3: `age: { min: 65 }`

Could handle `=` too with just `age: 30`

Reply:
Good catch - I've updated it and it's ready for review